contracts/estate/EstateRegistry.sol

```diff
@@ -372,4 +372,26 @@ contract EstateRegistry is Migratable, ERC721Token, ERC721Receiver, Ownable, IEs
 
     return uint256(out);
   }
+
+  /**
+   * @dev update LAND data owned by an Estate
+   * @param estateId Estate
+   * @param x coordinate of the LAND
+   * @param y coordinate of the LAND
+   * @param data string metadata
+   */
+  function updateLandData(uint256 estateId, int x, int y, string data) external onlyUpdateAuthorized(estateId) {
```

Comment:
Add a validation that `x` and `y` corresponds to a token within `estateId`, otherwise `onlyUpdateAuthorized` is not sufficient verification